app/views/missions/_form.html.erb

```diff
@@ -95,17 +100,11 @@
 
       </div>
       <% unless mission.new_record? %>
-        <div id="enrollments">
-          <%= f.fields_for :enrollments, mission.enrollments do |enrollment| %>
-            <%= render 'enrollment_fields', f: enrollment %>
-
-          <% end %>
-          <div class="links">
-            <%= link_to_add_association t(".enroll_another_member"),
-                                        f,
-                                        :enrollments %>
-          </div>
-        </div>
+        <% if mission.genre == "regulated"%>
+          <%= render 'enrollments_with_time_slots_form', f: f, mission: mission%>
+        <% else %>
+          <%= render 'enrollments_form', f: f, mission: mission %>
+        <% end %>
```

Comment:
The only thing that differs between `_enrollments_with_time_slots_form.html.erb` and `_enrollments_form.html.erb` partials, it the sub-partial that they call.
You can then simplify the partials logic by merging `_enrollments_with_time_slots_form.html.erb` and `_enrollments_form.html.erb` together, and pass the name of the sub-partial you want to render in this form.
E.g.:

```ruby
<% if mission.genre == "regulated" %>
  <%= render 'enrollments_form',
             f: f,
             mission: mission,
             cocoon_fields_name: 'enrollments_with_time_slots_field' %>
<% else %>
  <%= render 'enrollments_form',
             f: f,
             mission: mission,
             cocoon_fields_name: 'enrollments_fields' %>
<% end %>
```
and then using the `coccon_fields_name` variable in the enrollments_form accordingly